OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -739,6 +648,22 @@ public static void init(Context context, String googleProjectNumber, String oneS
       startPendingTasks();
    }
 
+   private static void setupActivityLifecycleListener(boolean wasAppContextNull) {
+      // Register the lifecycle listener of the app for state changes in activities with proper context
+      ActivityLifecycleListener.registerActivityLifecycleCallbacks((Application) appContext);
```

Comment:
Shouldn't this be inside of `if (wasAppContextNull)`? If so let's just remove the boolean param and just guard around where it was called.